src/measure_theory/set_integral.lean

```diff
@@ -0,0 +1,285 @@
+/-
+Copyright (c) 2020 Zhouhang Zhou. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Zhouhang Zhou
+-/
+
+import measure_theory.bochner_integration
+import measure_theory.indicator_function
+
+/-!
+# Set integral
+
+Integrate a function over a subset of a measure space.
+
+## Main definition
+
+`measurable_on`, `integrable_on`, `integral_on`
+
+## Tags
+
+indicator, characteristic
+-/
+
+noncomputable theory
+open_locale classical topological_space
+open set lattice filter topological_space ennreal emetric measure_theory
+
+set_option class.instance_max_depth 50
+
+universes u v w
+variables {α : Type u} {β : Type v} {γ : Type w}
+
+section measurable_on
+variables [measurable_space α] [measurable_space β] [has_zero β] {s : set α} {f : α → β}
+
+/-- `measurable_on s f` means `f` is measurable over the set `s`. -/
+@[reducible]
+def measurable_on (s : set α) (f : α → β) : Prop := measurable (indicator s f)
+
+lemma measurable_on_univ (hf : measurable f) : measurable_on univ f :=
+hf.if is_measurable.univ measurable_const
+
+lemma measurable_on_of_measurable (hs : is_measurable s) (hf : measurable f) : measurable_on s f :=
+hf.if hs measurable_const
```

Comment:
With name `measurable.measurable_on` it would be possible to use the dot notation.